app/controllers/foreman_tasks/tasks_controller.rb

```diff
@@ -62,8 +62,27 @@ def controller_name
 
     private
 
-    def find_task
-      ForemanTasks::Task::DynflowTask.find(params[:id])
+    def controller_permission
+      'foreman_tasks'
+    end
+
+    def action_permission
+      case params[:action]
+      when 'sub_tasks'
+        :view
+      when 'resume', 'unlock', 'force_unlock', 'cancel_step'
+        :edit
+      else
+        super
+      end
+    end
```

Comment:
@ares @iNecas mentioned that this may not be needed. How can we get rid of it? (I needed to add it too in foreman_deployments.)